### DISCUSSION TITLE
NumericUpDown rollover behavior

### DISCUSSION THREAD

**Discussion by dknippel** [MahApps/MahApps.Metro#4473] [Q&A] [answered]
Hi All,

Does the NumericUpDown have any built-in rollover behavior?

For example let's say we have the said control tied to an angle (degrees) property.  If the user increments said control to 180 degrees and then increments one more time, it would rollover to -179 degrees.

Thank you,
Dennis

**Comment by punker76** [comment]
@dknippel Did you found a solution? If yes, then you can maybe provide it here for other who are possible interested in too. 🙏

**Reply by dknippel** [reply, marked as answer]
We did find a solution, albeit specific to our particular scenario addressing degrees.  Essentially we added the following logic to the backing property to which said NumericUpDown control is bound to:
```
        private double _rotateX;
        public double RotateX
        {
            get { return _rotateX; }
            set 
            {
                value = (((value + 180d) % 360d) + 360d) % 360d - 180d;
                SetProp<double>(ref _rotateX, value);
            }
        }
```

Granted, I don't think this is ideal but hopefully it might help someone else out there or perhaps spark an improved solution.

In exchange, could perhaps someone help me with another discussion item #4433 ??? :-D

**Comment by dknippel** [comment]
Hi Jan,

Is there any help you could provide regarding my other post here:  https://github.com/MahApps/MahApps.Metro/discussions/4433

Thanks,

Dennis M. Knippel  |  Wiesenportz Enterprises, Inc.<https://urldefense.proofpoint.com/v2/url?u=http-3A__wiesenportz.com_&d=DwMFaQ&c=q6k2DsTcEGCcCb_WtVSz6hhIl8hvYssy7sH8ZwfbbKU&r=Ply5u_PjP6B9Q52jMvep4RJPeZlZqmgBzZLWUaWL76I&m=atuSafHOfyP3bD89wZOAicUrw8FwPnHLx2fTuqtR_mLQYovbvtUAvWM7i9_hkF6f&s=iETnAtNq17tk0Q_NOHK2Rr0PM7AElVOdsLu83HF7Fyk&e=>



1400 112th Avenue SE, Suite 100<https://urldefense.proofpoint.com/v2/url?u=https-3A__www.bing.com_maps-3Fq-3D1400-2B112th-2BAvenue-2BSE-252C-2BSuite-2B100-2Bbellevue-2Bwa-2B98004-26form-3DWNSGPH-26qs-3DSW-26cvid-3Dba18c22589384cfd965d226e25a29bcd-26pq-3D1400-2B112th-2BAvenue-2BSE-252C-2BSuite-2B100-2Bbellevue-2Bwa-2B98004-26cc-3DUS-26setlang-3Den-2DUS-26nclid-3D8F062EC1290E1CCA6D1237378CAF015B-26ts-3D1637692654026-26wsso-3DModerate&d=DwMFaQ&c=q6k2DsTcEGCcCb_WtVSz6hhIl8hvYssy7sH8ZwfbbKU&r=Ply5u_PjP6B9Q52jMvep4RJPeZlZqmgBzZLWUaWL76I&m=atuSafHOfyP3bD89wZOAicUrw8FwPnHLx2fTuqtR_mLQYovbvtUAvWM7i9_hkF6f&s=p0CF6uPtc4bBTg3hTprZNcnFEJSbG234CwK0nnAmWLo&e=>  |  Bellevue, WA  98004  |  425.301.1030  | ***@***.******@***.***>

________________________________
